appsdk/sdk.go

```diff
@@ -129,6 +125,7 @@ type AppFunctionsSDK struct {
 	deferredFunctions         []bootstrap.Deferred
 	serviceKeyOverride        string
 	backgroundChannel         <-chan types.MessageEnvelope
+	customTriggerBuilders     map[string]func(sdk *AppFunctionsSDK) (Trigger, error)
```

Comment:
how about `customTriggerFactories` instead since these are factory methods and it is more consistent with existing terms.